### PR TITLE
[fix] Improve ownership handling and documentation for Stacked Diagrams

### PR DESCRIPTION
**Pull request by gacarrillor**
Followup #58568 (probably the last one in this series)

Improves ownership handling of sub renderers in `QgsStackedDiagramRenderer` and `QgsStackedDiagramPropertiesModel`.

Adds documentation on ownership transfer when receiving or returning sub renderers.

Other adjustments:
 + Replaces `subDiagram` by `subDiagramRenderer` when we actually refer to renderers.
 + Avoids `auto` for non-complex types.
 + Passing from stacked to single diagram: take first stacked diagram as the basis for the new single one.
 + Stacked diagram list in GUI: Match `Diagram type` column to names from the main diagram combobox. This helps us to better differentiate between a stacked bars diagram and a stacked diagram (using the diagram name alone was a bit confusing: e.g., `Stacked`).

### CODE REVIEW

File: src/gui/vector/qgsstackeddiagramproperties.cpp [hunk 1]
```diff
@@ -189,7 +189,7 @@ void QgsStackedDiagramProperties::syncToLayer()
       const auto renderers = stackedDiagramRenderer->renderers();
       for ( const auto &renderer : renderers )
```
Comment (nyalldawson) [inline src/gui/vector/qgsstackeddiagramproperties.cpp:190]: Can you avoid the use of auto for things like this? It goes against current QGIS coding guidelines, which is only to use auto for complex types like iterators.

**Review by nyalldawson — APPROVED** [review]
Looks good, thanks!

File: src/gui/vector/qgsstackeddiagramproperties.cpp (hunk 1 above)
Comment (gacarrillor) [reply src/gui/vector/qgsstackeddiagramproperties.cpp:190]: Sure! Thanks for the suggestion.

**Comment by gacarrillor** [comment]
@nyalldawson, do you know how to get rid of the error in code layout?

```
ERROR: Issues in 'warning' category found:
/home/runner/work/QGIS/QGIS/src/core/qgsdiagramrenderer.cpp:867,warning,missingMemberCopy,Member variable 'QgsStackedDiagramRenderer::mDiagramRenderers' is not assigned in the copy constructor. Should it be copied?
```

**Comment by nyalldawson** [comment]
@gacarrillor 
> do you know how to get rid of the error in code layout?

Add an explicit empty constructor for  mDiagramRenderers -- that indicates that you're intentionally NOT copying that field

File: src/gui/vector/qgsstackeddiagramproperties.cpp (hunk 1 above)
Comment (gacarrillor) [reply src/gui/vector/qgsstackeddiagramproperties.cpp]: Addressed in https://github.com/qgis/QGIS/pull/59123/commits/3bbad5495b61cbc8bc297c01f6c8cce831ef8bd8

**Comment by gacarrillor** [comment]
@nyalldawson,
Sorry, I added two last commits to sort of close this followup series.
Thanks for your review!

**Pull request merged**